app/controllers/report_controller/saved_reports.rb

```diff
@@ -156,10 +156,10 @@ def set_saved_reports_condition(rep_id = nil)
 
     # Admin users can see all saved reports
     unless admin_user?
-      cond[0] << " AND miq_group_id=?"
-      cond.push(current_group_id)
+      cond[0] << " AND miq_group_id IN (?)"
+      cond.push(current_user.miq_groups.collect(&:id))
```

Comment:
@lpichler @gtanzillo Not in this pull request... but we should look at converting this method to return a `.where`... it looks like it's called in report_controller like this: `rep = MiqReportResult.where(set_saved_reports_condition).limit(1).order('created_on desc').pluck("created_on").first` where it's an easy change by just moving this code to MiqReportResult and have it return a relation.  

It's also passed to get_view, which might be a harder change to use a method on MiqReportResult that does a `.where`... `view, pages = get_view(MiqReportResult, :where_clause => set_saved_reports_condition(@sb[:miq_report_id]))`

It might be worth figuring out how much work this is to do though.